source/simulator/core.cc

```diff
@@ -2422,11 +2421,11 @@ namespace aspect
         // if so, then loop over as many times as this is necessary
         if ((parameters.additional_refinement_times.size() > 0)
             &&
-            (parameters.additional_refinement_times.front () < time+time_step))
+            (parameters.additional_refinement_times.front () < time+new_time_step))
           {
             while ((parameters.additional_refinement_times.size() > 0)
                    &&
-                   (parameters.additional_refinement_times.front () < time+time_step))
+                   (parameters.additional_refinement_times.front () < time+new_time_step))
```

Comment:
Should we be incrementing the time by the new timestep? If this is incorrect, then the time update is also incorrect.